### ISSUE TITLE
Paged attention single-request latency 2x slower than FlashInfer at batch=1

### ISSUE THREAD

**Issue by sunway513** [ROCm/aiter#2495]
## Summary
AITER `paged_attention_v1` has ~2x higher kernel launch latency than FlashInfer `BatchDecodeWithPagedKVCacheWrapper` at batch=1 on single-token decode. This impacts TTFT (Time to First Token) for low-concurrency serving.

## Measurements

**Llama3-8B, HQ=32, HKV=8, HD=128, SK=4096, page_size=16**

| Batch | AITER PA (MI355X) | FlashInfer (B300) | Ratio |
|---|---|---|---|
| 1 | 2.97 TFLOPS (0.023ms) | 5.83 TFLOPS (0.012ms) | **0.48x** |
| 4 | 11.45 TFLOPS (0.023ms) | — | — |
| 16 | 24.39 TFLOPS (0.044ms) | 17.30 TFLOPS (0.050ms) | **1.41x** |
| 64 | 21.08 TFLOPS (0.204ms) | 14.83 TFLOPS (0.290ms) | **1.42x** |

At B=1, absolute latency is 0.023ms vs 0.012ms — the gap is small in absolute terms but shows up as 2x in TFLOPS ratio. At B>=16, AITER PA is faster.

**Consistent across models at B=1 SK=4096:**

| Model | HQ | HKV | AITER (TFLOPS) | FlashInfer (TFLOPS) | Ratio |
|---|---|---|---|---|---|
| Llama3-8B | 32 | 8 | 2.82 | 5.83 | 0.48x |
| Mistral-Small-4 | 32 | 32 | 1.52 | 5.22 | 0.29x |
| Qwen3-Coder-480B | 96 | 8 | 2.98 | 10.26 | 0.29x |
| Step-3.5-Flash | 64 | 8 | 2.97 | 9.21 | 0.32x |
| GLM-5 | 64 | 64 | 1.50 | 4.67 | 0.32x |

## Analysis
- At B=1, total compute is ~0.01ms of useful work — the rest is kernel launch overhead
- FlashInfer appears to have lower launch overhead, possibly due to:
  - CUDA graph integration (`CUDAGraphBatchDecodeWithPagedKVCacheWrapper`)
  - Persistent kernel design
  - Lower host-device synchronization cost
- MHA configs (HQ=HKV) are worst — AITER PA seems to scale per-head rather than batch

## Impact
- Affects TTFT metric in production serving at low concurrency
- B=1 decode is the critical path for interactive chat (single user)
- At high concurrency (B>=16), AITER PA is competitive or faster

## Reproduce
```python
from aiter import paged_attention_v1
import torch, time

batch, hq, hkv, hd, sk, ps = 1, 32, 8, 128, 4096, 16
npps = sk // ps; total = batch * npps
q = torch.randn(batch, hq, hd, dtype=torch.bfloat16, device='cuda')
kc = torch.randn(total, ps, hkv, hd, dtype=torch.bfloat16, device='cuda')
vc = torch.randn(total, ps, hkv, hd, dtype=torch.bfloat16, device='cuda')
bt = torch.arange(total, device='cuda', dtype=torch.int32).reshape(batch, npps)
sl = torch.full((batch,), sk, device='cuda', dtype=torch.int32)
out = torch.empty(batch, hq, hd, dtype=torch.bfloat16, device='cuda')
mp = (sk+255)//256
ws = torch.empty(batch, hq, mp, hd+1, dtype=torch.float32, device='cuda')
ks = torch.tensor(1.0, dtype=torch.float32, device='cuda')
vs = torch.tensor(1.0, dtype=torch.float32, device='cuda')

def run():
    paged_attention_v1(out, ws, q, kc, vc, hd**-0.5, bt, None, sl, sk, None, 'auto', 'NHD', 0.0, ks, vs)

[run() for _ in range(50)]
torch.cuda.synchronize(); t=time.perf_counter()
[run() for _ in range(200)]
torch.cuda.synchronize(); ms=(time.perf_counter()-t)/200*1000
print(f'B=1: {4*batch*hq*sk*hd/ms*1e-9:.2f} TFLOPS ({ms:.4f}ms)')
```

Tested on MI355X with `rocm/atom-dev:latest` container.

cc @sunway513
